pkg/crd/generator/generator.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generator
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	extensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/gengo/args"
+	crdutil "sigs.k8s.io/controller-tools/pkg/crd/util"
+	"sigs.k8s.io/controller-tools/pkg/internal/codegen"
+	"sigs.k8s.io/controller-tools/pkg/internal/codegen/parse"
+	"sigs.k8s.io/controller-tools/pkg/util"
+)
+
+// Generator search API resource definition files,
+// and generate CRD install files accordingly.
+type Generator struct {
+	RootPath          string
+	Domain            string
+	Namespace         string
+	SkipMapValidation bool
+}
```

Comment:
We should make output path also configurable ? and can default to `config/crds` if not specified ?